packages/core/backend/sirius-components-collaborative/src/main/java/org/eclipse/sirius/components/collaborative/api/IRepresentationEventProcessor.java

```diff
@@ -26,7 +26,7 @@
  *
  * @author sbegaudeau
  */
-public interface IRepresentationEventProcessor extends IDisposablePublisher {
+public interface IRepresentationEventProcessor extends IDisposablePublisher, IRepresentationEventProcessorFluxCustomizer {
```

Comment:
Re-read the code sample I gave you, the goal was not to have `IRepresentationEventProcessor` implement this interface but instead to have `org.eclipse.sirius.web.starter.SiriusWebStarterConfiguration#eventProcessorSubscriptionProvider` leverage it. And now that I rethink about it, the code should probably be more something like this:

```
    @Bean
    @ConditionalOnMissingBean
    public IEventProcessorSubscriptionProvider eventProcessorSubscriptionProvider(IEditingContextEventProcessorRegistry editingContextEventProcessorRegistry) {
        return new IEventProcessorSubscriptionProvider() {
            @Override
            public <T extends IRepresentationEventProcessor> Flux<IPayload> getSubscription(String editingContextId, Class<T> representationEventProcessorClass, IRepresentationConfiguration representationConfiguration, IInput input) {
                return editingContextEventProcessorRegistry.getOrCreateEditingContextEventProcessor(editingContextId)
                        .flatMap(processor -> processor.acquireRepresentationEventProcessor(representationEventProcessorClass, representationConfiguration, input))
                        .map(representationEventProcessor -> this.representationEventProcessorFluxCustomizer.customize(editingContextId, representationEventProcessor, representationEventProcessor.getOutputEvents(input)))
                        .orElse(Flux.empty());
            }
        };
    }
```

with

```
public interface IRepresentationEventProcessorFluxCustomizer {
    Flux<IPayload> customize(String editingContextId, IRepresentationEventProcessor representationEventProcessor, Flux<IPayload> outputEvents);
}
```

By giving access to the editing context id and the event processor, consumers of Sirius Web will be able to determine if they want to do something about a specific representation or not. We could even modify the code a tiny bit more with a canHandle method to support multiple customizers and pick the first one (all those?) supported for a specific representation...